### ISSUE TITLE
Adding support for DigitalOcean

### ISSUE THREAD

**Issue by demchris** [EngineerBetter/concourse-up#102]
Hi,

I would like to contribute to the project to add DigitalOcean as Teraform iaas provider to give the option to deploy concourse on DigitalOcean with the tool.

Is somebody able to give me hints, rules to follow to start on. Since Teraform supports DigitalOcean as provider, my first guess it's possible, may be it's not, I would like to ear from you guys about the idea.

Thank you,

Christian

**Comment by DanielJonesEB** [comment]
Hi @demchris,

Thanks for getting in touch. It'd be great to hear about your use case for DigitalOcean, so we can figure out how to prioritise such work.

The IAAS abstraction in Concourse-Up was done at high velocity, accruing a lot of technical debt. Now we've got GCP support implemented, we'd like to double-down on paying off that technical debt before adding any IAAS providers.

We'd love for folks to implement support themselves via PR, but to be quite honest I wouldn't recommend wading into the code to do this. I'm ultimately responsible for it, and even _I_ think it sucks.

We've committed previously to implementing Azure support, and as we do that we'll hopefully simplify the abstractions to the point where they're sane enough for other folks to build upon them.

**Comment by demchris** [comment]
Hi Daniel,

Thanks for your quick answer.

Perfect, I understand and totally agree with the approach. And of course, Azure must be priorized against DO.

Best regards,

Christian

Envoyé par Blue<http://www.bluemail.me/r?b=14667>
Le 18 mars 2019, à 04 h 49, Daniel Jones <notifications@github.com<mailto:notifications@github.com>> a écrit:

Hi @demchris<https://github.com/demchris>,

Thanks for getting in touch. It'd be great to hear about your use case for DigitalOcean, so we can figure out how to prioritise such work.

The IAAS abstraction in Concourse-Up was done at high velocity, accruing a lot of technical debt. Now we've got GCP support implemented, we'd like to double-down on paying off that technical debt before adding any IAAS providers.

We'd love for folks to implement support themselves via PR, but to be quite honest I wouldn't recommend wading into the code to do this. I'm ultimately responsible for it, and even I think it sucks.

We've committed previously to implementing Azure support, and as we do that we'll hopefully simplify the abstractions to the point where they're sane enough for other folks to build upon them.
